src/rust/ide/view/graph-editor/src/lib.rs

```diff
@@ -3090,6 +3267,57 @@ fn new_graph_editor(app:&Application) -> GraphEditor {
         eval quick_visualization_preview((value) model.nodes.set_quick_preview(*value));
     }
 
+
+
+    // =============
+    // === Modes ===
+    // =============
+
+    let profiling_mode_transition = Animation::new(network);
+    frp::extend! { network
+        out.source.mode <+ frp.toggle_profiling_mode.map2(&frp.mode,|_,&mode| {
+            match mode {
+                Mode::Profiling => Mode::Normal,
+                _               => Mode::Profiling,
+            }
+        });
+
+        eval out.mode([model](_) {
+            for edge_id in model.edges.keys() {
+                model.refresh_edge_color(edge_id);
+            }
+        });
+
+        profiling_mode_transition.target <+ out.mode.map(|&mode| {
+            match mode {
+                Mode::Normal    => 0.0,
+                Mode::Profiling => 1.0,
+            }
+        });
+        eval profiling_mode_transition.value ((&v) scene.dom.layers.back.set_grayscale(v));
+    }
+
+
+    // === Profiling Button ===
+
+    frp::extend! { network
+        out.source.mode <+ model.profiling_button.state.map(|&toggled| {
+            if toggled { Mode::Profiling } else { Mode::Normal }
+        });
+
+        model.profiling_button.set_state <+ out.mode.on_change().map(|&mode| {
+            matches!(mode,Mode::Profiling)
```

Comment:
In such cases add methods like `mode.is_profiling()` - it is easier to use and very useful in many places.